### ISSUE TITLE
Examples: Be able to use CentOS / RHEL / other operating systems for `podman machine init / start`

### ISSUE THREAD

**Issue by cdrage** [containers/podman-desktop#1638]
### Is your feature request related to a problem? Please describe

When I run podman machine, maybe I'd like to try using RHEL or other operating systems. Instead, I'm stuck using fedora-coreos / running the default podman machine operating system. 

### Describe the solution you'd like

We should be able to provide instructions / alternatives to try out Podman Desktop against RHEL, RHED, etc.

**Comment by ashley-cui** [comment]
I'd do this with caution, because at the current moment the Podman team can only support FCOS (eventually maybe RHELCOS), and doing a demo on this may give of the impression that we can support other OS's.

**Comment by baude** [comment]
agree. of all the things required rn, are there not significantly more impactful and import things to do ?

**Comment by afbjorklund** [comment]
You can use the existing Lima provider, to run other operating systems - such as Fedora (Cloud)

The instance name is currently hardcoded ("podman-lima"), but you can use any `template://`

EDIT: In v1.0.1, the hardcoded instance name was changed from "podman-lima" to just "podman".

`limactl start template://podman`

EDIT: In v1.1.0, it is not hardcoded anymore - but available in preferences 

----

If you choose another template (than podman), you have to install `podman` on the machine yourself

You can see how podman does it: https://github.com/lima-vm/lima/blob/master/examples/podman.yaml

And follow the upstream podman instructions: https://podman.io/docs/installation#installing-on-linux

`limactl start --name podman template://centos-stream`

**Comment by afbjorklund** [comment]
### Lima

Using CentOS Stream instead of the default Fedora Cloud, causes the image download to be **twice** as big:

```
475M	https://download.fedoraproject.org/pub/fedora/linux/releases/38/Cloud/x86_64/images/Fedora-Cloud-Base-38-1.6.x86_64.qcow2
1.1G	https://cloud.centos.org/centos/9-stream/x86_64/images/CentOS-Stream-GenericCloud-9-20230523.0.x86_64.qcow2
```

I'm not sure why this is, but it is still smaller than CoreOS (the download is 627M, but decompresses to 1.6G!)

```
646M	.local/share/containers/podman/machine/qemu/fedora-coreos-38.20230806.2.0-qemu.x86_64.qcow2.xz
1,6G	.local/share/containers/podman/machine/qemu/podman-machine-default_fedora-coreos-38.20230918.2.0-qemu.x86_64.qcow2
```

No public templates available for RHEL

But these distributions are available:

* AlmaLinux `template://almalinux`

* Rocky Linux `template://rocky`

----

There is also Alpine.

<details>
There is an Alpine image available as an alternative, it weighs in at 98M - but it doesn't do rootless (no systemd)

https://github.com/afbjorklund/lima/blob/nerdcore/examples/alpine-podman.yaml (currently slightly out of date)

It is just the `template://alpine`, with the latest `podman` pre-installed from the latest distro package available:

https://pkgs.alpinelinux.org/packages?name=podman&branch=edge&repo=community (`apk add podman`)

EDIT: Here are some ready-to-run examples, using the regular ISO:

https://github.com/afbjorklund/lima/tree/alpine-examples/examples
</details>

**Comment by afbjorklund** [comment]
Added template for Fedora CoreOS to Lima, adding ignition support to the cloud-init support:

* https://github.com/lima-vm/lima/pull/1653

`limactl start --name podman template://fedora-coreos`

`limactl start --name docker template://fedora-coreos`

It doesn't have the portForward by default, so that YAML needs to be added to the configuration.

But all of `podman` and `docker` and `nerdctl` are already installed, by the default installation here.

```yaml
portForwards:
- guestSocket: "/run/user/{{.UID}}/podman/podman.sock"
  hostSocket: "{{.Dir}}/sock/podman.sock"
```

**Comment by afbjorklund** [comment]
More examples, for Ubuntu and AlmaLinux: (you can probably make your own "remix")

https://github.com/afbjorklund/lima/tree/podman-desktop/examples/podman-desktop
